syntax = "proto3";

package million.ss.cluster;

message MsgBody {
    oneof body {
        HandshakeReq handshake_req = 1;
        HandshakeRes handshake_res = 2;
        ServiceQueryReq service_query_req = 3;
        ServiceQueryRes service_query_res = 4;
        ClusterSend cluster_send = 5;
        ClusterCall cluster_call = 6;
        ClusterReply cluster_reply = 7;
    }
}

message HandshakeReq {
    uint64 src_node_id = 1;
}

message HandshakeRes {
    uint64 target_node_id = 1;
}

message ServiceQueryReq {
    string target_service_name = 1;
}

message ServiceQueryRes {
    string target_service_name = 1;
    uint64 target_service_id = 2;
}

message ClusterSend {
    uint64 src_service_id = 1;
    uint64 target_service_id = 2;
    uint64 session_id = 3;
}

message ClusterCall {
    uint64 src_service_id = 1;
    uint64 target_service_id = 2;
    uint64 session_id = 3;
}

message ClusterReply {
    uint64 src_service_id = 1;
    uint64 target_service_id = 2;
    uint64 session_id = 3;
}